sycl/source/detail/program_impl.cpp

```diff
@@ -247,7 +247,7 @@ void program_impl::build_with_kernel_name(std::string KernelName,
     MBuildOptions = BuildOptions;
     MProgram = ProgramManager::getInstance().getBuiltPIProgram(
         detail::getSyclObjImpl(get_context()),
-        detail::getSyclObjImpl(get_devices()[0]), KernelName, this,
+        detail::getSyclObjImpl(get_devices()[0]), KernelName,
```

Comment:
Since this change assumes that `build_with_kernel_name` is no longer called, can we just remove it in this PR?